DataParsers/JsonWeatherDataParser.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Newtonsoft.Json;
+using Weather_Monitoring_System.Models;
+
+namespace Weather_Monitoring_System.DataParsers
+{
+    public class JsonWeatherDataParser : IWeatherDataParser
+    {
+        public WeatherData? TryParse(string data)
+        {
+            try
+            {
+                return JsonConvert.DeserializeObject<WeatherData>(data);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                Environment.Exit(1);
+                return null;
+            }
+
+        }
+    }
```

Comment:
```suggestion
        public bool TryParse(string data, out WeatherData? weatherData)
        {
           weatherData = null;
            try
            {
                weatherData = JsonConvert.DeserializeObject<WeatherData>(data);
                return true;
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                Environment.Exit(1);
                return false;
            }

        }
    }
```

this is how we usually deal with the Try thing methods.